syncd/syncd_saiswitch.cpp

```diff
@@ -1185,6 +1221,142 @@ sai_object_id_t SaiSwitch::getDefaultValueForOidAttr(
     return ita->second;
 }
 
+std::map<sai_object_id_t, std::vector<sai_object_id_t>> SaiSwitch::saiGetQueueMap(const std::vector<sai_object_id_t> &port_list) const
+{
+    SWSS_LOG_ENTER();
+
+    std::map<sai_object_id_t, std::vector<sai_object_id_t>> queue_map;
+
+    for (auto &port_rid : port_list)
+    {
+        sai_attribute_t sai_attr;
+        sai_attr.id = SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES;
+        sai_status_t status = sai_metadata_sai_port_api->get_port_attribute(port_rid, 1, &sai_attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("failed to collect number of queues for port RID %s: %s",
+                    sai_serialize_object_id(port_rid).c_str(),
+                    sai_serialize_status(status).c_str());
+
+            continue;
+        }
+
+        std::vector<sai_object_id_t> queue_list(sai_attr.value.u32);
+
+        sai_attr.id = SAI_PORT_ATTR_QOS_QUEUE_LIST;
+        sai_attr.value.objlist.list = queue_list.data();
+        status = sai_metadata_sai_port_api->get_port_attribute(port_rid, 1, &sai_attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("failed to collect queues list for port RID %s: %s",
+                    sai_serialize_object_id(port_rid).c_str(),
+                    sai_serialize_status(status).c_str());
+
+            continue;
+        }
+
+        queue_map[port_rid] = queue_list;
+    }
+
+    return queue_map;
+}
+
+void SaiSwitch::initQueueMap()
+{
+    uint32_t countersSize = (uint32_t)m_supported_queue_counters.size();
+    std::vector<uint64_t> counters;
+    counters.resize(countersSize);
+
+    swss::DBConnector db(COUNTERS_DB, swss::DBConnector::DEFAULT_UNIXSOCKET, 0);
+    swss::Table countersTable(&db, "COUNTERS");
+    sai_status_t status;
+
+    for (auto &port_queues: m_queue_rid_map)
+    {
+        for (auto queue_it = port_queues.second.begin(); queue_it != port_queues.second.end();)
```

Comment:
>; [](start = 93, length = 1)

you do not need semicolon here.

Reply:
actually needed. I suggest rewrite as while-loop for readability.

---
In reply to: [143101748](https://github.com/Azure/sonic-sairedis/pull/230#discussion_r143101748) [](ancestors = 143101748)